src/components/realm.cairo

```diff
@@ -0,0 +1,59 @@
+use array::ArrayTrait;
+use traits::Into;
+use eternum::alias::ID;
+use starknet::ContractAddress;
+use option::OptionTrait;
+use eternum::constants::RESOURCE_IDS_PACKED_SIZE;
+use eternum::constants::PRIME;
+use traits::BitAnd;
+use eternum::utils::unpack::unpack_resource_ids;
+use debug::PrintTrait;
+
+#[derive(Component)]
+struct Realm {
+    realm_id: ID, // OG Realm Id
+    // TODO: no need for owner ? since we use Owner component
+    owner: ContractAddress,
+    // packed resource ids of realm
+    resource_ids_packed: u128, // max 16 resources
+    resource_ids_count: u8,
+    cities: u8,
+    harbors: u8,
+    rivers: u8,
+    regions: u8,
+    wonder: u8,
+    order: u8
+}
+
+
+trait RealmTrait {
+    fn is_owner(self: Realm, address: ContractAddress) -> bool;
+    fn has_resource(self: Realm, resource_id: u8) -> bool;
+}
+
+impl RealmImpl of RealmTrait {
+    fn is_owner(self: Realm, address: ContractAddress) -> bool {
+        if (self.owner == address) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    fn has_resource(self: Realm, resource_id: u8) -> bool {
+        let mut resource_ids: Array<u8> = unpack_resource_ids(
```

Comment:
Wouldn't it be better if `unpack_resource_ids` is a method on self? Does it make sense outside of a Realm component?

Now I'm even starting to think it would be cool to have some kind of lazily loaded component members - I can imagine `has_resource` can be called multiple times in a single TX, which would call `unpack_resource_ids` N times as well, but it's really only necessary to do it once. Not sure if it's possible yet in Cairo/Dojo, but an intriguing idea nonetheless. 

Also, in any case, the `unpack_resource_ids` should IMO return a `Span<u8>`, not an array, as we don't need to modify it after unpacking, do we?

Reply:
thought about this as well but i'm using this in other parts as well for ```LaborCR``` and ```BuildingTypeConfig``` components.
Love the idea as well but not sure if possible in Cairo. In that case i would maybe split the function into two :
1. ```get_resources``` retrieves array
2. ```has_resources``` takes array as input

Agreed on the span !